aldryn_newsblog/models.py

```diff
@@ -235,7 +235,7 @@ def save(self, *args, **kwargs):
 
         # Ensure we aren't colliding with an existing slug *for this language*.
         if not Article.objects.translated(
-                slug=self.slug).exclude(pk=self.pk).exists():
+                slug=force_unicode(self.slug)).exclude(pk=self.pk).exists():
```

Comment:
> This needs to be force_text()

I checked the imports here and followed those: https://github.com/aldryn/aldryn-newsblog/blob/fix/empty-lead-in-handling/aldryn_newsblog/utils/utilities.py#L11-L14

> but I don't think it should be here at all, TBH. NewsBlog when used normally should never have lazy objects here.

In a perfect world, yes :smiley: This is a design question/decision really, but I think the reusable apps should be rather forgiving than strict on those sorts of things (three years of Java in school were enough :stuck_out_tongue_winking_eye:)